public/js/monitor.js

```diff
@@ -1,90 +1,125 @@
-    "foundBreaches": "User scanned additional email.",
-    "noBreaches": "User scanned email address after scanning email with no associated breaches.",
-    "landingPage": "User submitted email from landing page.",
-
+    "eventAction": "",
+  },
+  "ShowAdditional": {
+    "eventCategory": "Show Additional Breaches",
+    "eventAction": "Button Click",
   },
   "Pageview": {
     "eventCategory":"Views",
-    "eventAction": "Viewed",
-    "featuredBreach": "Featured Breach",
-    "foundBreaches": "List of compromised accounts.",
-    "noBreaches": "No associated breaches.",
-    "landingPage": "Default landing page without featured breach.",
-    "confirmAccount": "Your account is confirmed page.",
+    "eventAction": "",
+  },
+  "Download": {
+    "eventCategory": "Download",
+    "eventAction": "Firefox Download",
+  },
+  "Link": {
+    "eventCategory": "Link Clicks",
+    "eventAction": "Link Click",
   },
 };
 
+// determines which page the event occurs and is sent to google analytics in ga_sendPing() as the eventLabel or custom dimension.
+
 function ga_getLocation() {
-  if (document.getElementById("featured-breach-landing")) {
-      return("featuredBreach");
+  if (document.querySelector(".landing-content")) {
+    if (document.getElementById("featured-breach-landing")) {
+      return "Featured Breach Page";
+    }
+    return "Landing Page";
   }
   if (document.getElementById("found-breaches")) {
-      return("foundBreaches");
+      return "Scan Results - found breaches";
   }
   if (document.getElementById("no-breaches")) {
-      return("noBreaches");
+      return "Scan Results - no breaches";
+  }
+  if (document.getElementById("unsubscribe-form")) {
+    return "Unsubscribe Page";
   }
   if (document.getElementById("unsubscribe-survey-form")) {
-    return("unsubscribePage");
+    return "Unsubscribe Survey";
+  }
+  if (document.getElementById("confirmation")) {
+    return "Account Confirmation Page";
+  }
+  if (document.getElementById("error-message")) {
+    return "Error Page";
   }
-    return("landingPage");
 }
 
-function ga_sendPing(string, eventLabel = false) {
-  if(typeof(ga) !== "undefined") {
 
-    const event = events[string];
+function ga_sendPing(eventDescription, eventLabel) {
+  if (typeof(ga) !== "undefined") {
+    const event = eventList[eventDescription];
     const eventCategory = event["eventCategory"];
     const eventAction = event["eventAction"];
 
-    if (!eventLabel) {
-      const eventLocation = ga_getLocation();
-      eventLabel = event[eventLocation];
-      ga("send", "event", eventCategory, eventAction, eventLabel);
-      return;
+    if (eventLabel) {
+      if (eventDescription.includes("UnsubscribeSurvey")) {
+        return ga("send", "event", eventCategory, eventAction, eventLabel);
+      }
+      if (eventDescription.includes("Download")) {
+        return ga("send", "event", eventCategory, eventAction, eventLabel, "", {"dimension1": ga_getLocation()});
```

Comment:
Is the location supposed to go in a different dimension for each category? Or is it always supposed to be in dimension1 ?

Reply:
The location, or other value, goes in a unique dimension for each event category where eventLabel is already in use. So in the case of error pages, eventLabel = 'Error Page' and the value sent in 'dimension4' is the error-message for that error page. 
<img width="1213" alt="screen shot 2018-09-23 at 5 49 51 pm" src="https://user-images.githubusercontent.com/22355127/45933797-8fc75500-bf59-11e8-90ea-710ee3cf8c64.png">